### PR TITLE
Deprecate SbStringFormat functions

### PR DESCRIPTION
**Pull request by maxz-lab**
Replace SbStringFormat*() calls with POSIX vsnprintf().
  SbStringFormat        -> vsnprintf
  SbStringFormatF       -> snprintf
  SbStringFormatWide    -> vswprintf
  SbStringFormatUnsafeF -> sprintf

Backward compatibility is protected by "SB_API_VERSION < 16".

Tested on Linux, Android, Evergreen(14,15,16), and Win32.

b/307941391

Change-Id: I437ff3d7bf735b46767867ea844aeffb28e58b4d

### CODE REVIEW

**Comment by andrewsavage1** [comment]
Are you waiting for a review from me? Once you are, please remove WIP from the title of the PR

**Comment by kaidokert** [comment]
I recommend converting this to draft for now, until majority builds pass.

**Comment by maxz-lab** [comment]
> Are you waiting for a review from me? Once you are, please remove WIP from the title of the PR

I think it is ready to be reviewed now.
Apologize for adding you in a bit too soon. I actually did not know how it happened.

File: starboard/nplb/string_format_test.cc
Comment (y4vor) [inline starboard/nplb/string_format_test.cc:28]: for the NPLB tests let's keep the original code but just guard the whole test with #if SB_API_VERSION < 16

File: starboard/nplb/string_format_wide_test.cc
Comment (y4vor) [inline starboard/nplb/string_format_wide_test.cc:26]: Same here

File: starboard/shared/stub/string_format.cc
Comment (y4vor) [inline starboard/shared/stub/string_format.cc:17]: The conversion is to use #if SB_API_VERSION < 16. Please update here and the other places.

File: third_party/musl/src/stdio/vswprintf.c
Comment (y4vor) [inline third_party/musl/src/stdio/vswprintf.c:38]: remove this check

File: third_party/v8/src/base/platform/platform-starboard.cc
Comment (y4vor) [inline third_party/v8/src/base/platform/platform-starboard.cc:349]: get rid of SbStringFormat

File: starboard/nplb/string_format_test.cc
Comment (y4vor) [reply starboard/nplb/string_format_test.cc:28]: Please bring the check at the top of the file and don't change the symbol.

For NPLB we don't change the API we just turn off the whole test for SB_API_VERSION < 16.

File: starboard/nplb/string_format_wide_test.cc
Comment (y4vor) [inline starboard/nplb/string_format_wide_test.cc:32]: Take a look at the other NPLB test comment.

**Review by y4vor — COMMENTED** [review]
Please update starboard/CHANGELOG.md

File: starboard/nplb/string_format_wide_test.cc
Comment (y4vor) [inline starboard/nplb/string_format_wide_test.cc:32]: keep SbStringFormatWide.

File: base/strings/string_util_starboard.h
Comment (andrewsavage1) [inline base/strings/string_util_starboard.h:1]: Are we able to fully replace this file with string_util_posix?

File: base/strings/stringprintf.cc
Comment (andrewsavage1) [inline base/strings/stringprintf.cc:32]: We shouldn't be changing things like this, as this is in upstream as well. Calling base::vsnprintf is preferable

File: third_party/mini_chromium/base/strings/stringprintf.cc
Comment (andrewsavage1) [inline third_party/mini_chromium/base/strings/stringprintf.cc:24]: This should use base::vsnprintf and should be fine without more changes

**Review by andrewsavage1 — COMMENTED** [review]
Please make sure to test internally as well

File: starboard/string.h
Comment (maxz-lab) [reply starboard/string.h:25]: It is needed because essentially older SB versions switch to stdio implementation as well.

File: starboard/string.h
Comment (maxz-lab) [reply starboard/string.h:102]: You are right. These should be kept as it was, i.e. SbStringFormat*() for older SB versions.

File: third_party/protobuf/src/google/protobuf/stubs/strutil.cc
Comment (maxz-lab) [reply third_party/protobuf/src/google/protobuf/stubs/strutil.cc:42]: We can clean up the code later afterward, since replacing API is more critical at this moment.

File: base/strings/stringprintf.cc
Comment (maxz-lab) [reply base/strings/stringprintf.cc:32]: This change is needed.

File: base/strings/string_util_starboard.h
Comment (maxz-lab) [reply base/strings/string_util_starboard.h:1]: I propose to clean up the code after this API replacement CL get merged.

File: base/strings/stringprintf.cc
Comment (andrewsavage1) [reply base/strings/stringprintf.cc:32]: Could you please explain why this is needed instead of just saying it is? It seems to me that base::vsnprintf is there for a reason given the comment in `base/strings/string_util.h`

File: starboard/tools/api_leak_detector/api_leak_detector.py
Comment (andrewsavage1) [inline starboard/tools/api_leak_detector/api_leak_detector.py:97]: Why do you add the symbols here if they're already added in c99.md? They shouldn't be necessary in both places

Comment (maxz-lab) [reply starboard/tools/api_leak_detector/api_leak_detector.py]: Good catch!

File: base/strings/stringprintf.cc
Comment (maxz-lab) [reply base/strings/stringprintf.cc:32]: The declaration in base/strings/string_util.h has been removed in this CL.

Comment (maxz-lab) [reply base/strings/stringprintf.cc]: Yes. It is updated.

File: base/strings/string_util.h
Comment (andrewsavage1) [inline base/strings/string_util.h:53]: This shouldn't be prefixed with `::` as we want it to call base::vsnprintf

File: base/strings/string_util.h
Comment (andrewsavage1) [inline base/strings/string_util.h:13]: This should be removed as there will be no references to stl's vsnprintf here

File: base/strings/string_util_starboard.h
Comment (andrewsavage1) [inline base/strings/string_util_starboard.h:30]: We should include stdio.h in SB 16+ for this file

File: base/strings/string_util.h
Comment (maxz-lab) [reply base/strings/string_util.h]: Fixed.

File: base/strings/string_util_starboard.h
Comment (maxz-lab) [reply base/strings/string_util_starboard.h:30]: Fixed.

Comment (andrewsavage1) [reply base/strings/string_util_starboard.h:30]: It was? stdio.h is still included in string_util.h but not in this file from what I can see

Comment (maxz-lab) [reply base/strings/string_util_starboard.h:33]: Added the inline function and missed the header. My bad.

File: base/strings/string_util.h
Comment (andrewsavage1) [reply base/strings/string_util.h:12]: This should be removed

File: base/strings/string_util_starboard.h
Comment (andrewsavage1) [reply base/strings/string_util_starboard.h:33]: Great, thank you

File: base/strings/string_util.h
Comment (andrewsavage1) [inline base/strings/string_util.h:50]: Just use vsnprintf, it will resolve to base::vsnprintf and be consistent with the upstream code

**Review by andrewsavage1 — APPROVED** [review]

**Pull request merged**
